Assets/Scripts/Score/ScoreManager.cs

```diff
@@ -44,35 +38,35 @@ public void UpdateScoreDisplay()
 
     public void LevelEnd()
     {
+        int pastHighScore = PlayerPrefs.GetInt(sceneHighScoreKey);
+
         if (playerTransform.position.y > startPlayerYPos)
         {
-            ChangeScore(playerDistPtsMultiplier * (playerTransform.position.y - startPlayerYPos));
+            ChangeScore(playerDistPtsMultiplier * (int) (playerTransform.position.y - startPlayerYPos));
         }
 
-        Score = (pastHighScore > Score) ? pastHighScore : Score;
-        PlayerPrefs.SetFloat(sceneHighScore, Score);
+        if (score > pastHighScore) { PlayerPrefs.SetInt(sceneHighScoreKey, score); }
     }
 
     private void Awake()
     {
         Instance = this;
-        startPlayerYPos = playerTransform.position.y;
+        startPlayerYPos = (int) playerTransform.position.y;
         int buildIndex = SceneManager.GetActiveScene().buildIndex;
 
         if (buildIndex == 0)
         {
-            sceneHighScore = "lvlIHighScore";
+            sceneHighScoreKey = "lvlIHighScore";
```

Comment:
I think this key should be changed to `lvlInfHighScore` or `levelInfiniteHighScore` since `lvlI` is less clear and could be mistaken for level 1.